Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -525,142 +525,56 @@ public class CameraManager {
+
         guard let mapView = mapView else {
-            assertionFailure("MapView is nil.")
-            completion?(false)
-            return
+            return nil
         }
 
-        guard let flyTo = FlyToInterpolator(from: mapView.cameraView.camera,
-                                            to: camera,
-                                            size: mapView.bounds.size) else {
-            completion?(false)
-            assertionFailure("FlyToInterpolator could not be created.")
-            return
+        guard let interpolator = FlyToInterpolator(from: mapView.cameraView.camera,
+                                                   to: camera,
+                                                   size: mapView.bounds.size) else {
+            return nil
         }
 
-        var time = duration ?? -1.0
-
         // If there was no duration specified, or a negative argument, use a default
-        if time < 0.0 {
-            time = flyTo.duration()
-        }
-
-        guard time > 0.0 else {
-            setCamera(to: camera, completion: completion)
-            return
-        }
+        let time: TimeInterval = duration ?? interpolator.duration()
 
         // TODO: Consider timesteps based on the flyTo curve, for example, it would be beneficial to have a higher
         // density of time steps at towards the start and end of the animation to avoid jiggling.
         let timeSteps = stride(from: 0.0, through: 1.0, by: 0.025)
-        let keyTimes: [NSNumber] = Array(timeSteps).map {
-            NSNumber(value: $0)
-        }
-
-        var animations: [CAAnimation] = []
+        let keyTimes: [Double] = Array(timeSteps)
 
-        if camera.zoom != nil {
-            let zoomLevels: [Double] = keyTimes.map { (number) -> Double in
-                return flyTo.zoom(at: number.doubleValue)
-            }
+        let animator = makeCameraAnimator(duration: time, curve: .linear) {
 
-            let zoomAnimation = CAKeyframeAnimation(keyPath: "zoom")
-            zoomAnimation.keyTimes              = keyTimes
-            zoomAnimation.values                = zoomLevels
-            zoomAnimation.duration              = time
-            zoomAnimation.isAdditive            = false
-            zoomAnimation.calculationMode       = .cubic
-            zoomAnimation.isRemovedOnCompletion = false
-            zoomAnimation.fillMode              = .forwards
+            UIView.animateKeyframes(withDuration: 0, delay: 0, options: []) {
 
-            animations.append(zoomAnimation)
-        }
+                for keyTime in keyTimes {
+                    let interpolatedCoordinate = interpolator.coordinate(at: keyTime)
+                    let interpolatedZoom = interpolator.zoom(at: keyTime)
+                    let interpolatedBearing = interpolator.bearing(at: keyTime)
+                    let interpolatedPitch = interpolator.pitch(at: keyTime)
```

Comment:
Why not also set the `interpolatedPadding`?

Reply:
When I tested that, I experienced a lot of jitter in the animation. So I opted to leave it out

Reply:
but you're completely ignoring it -- it's not being set at the end of the animation even.

Reply:
Correct, but if you look at the existing V1 animations, the same was happening.

Reply:
Hmm let's revisit once this lands.